### ISSUE TITLE
Design and implement post-rendering autoformatting

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#36]
### TL;DR

As a template user, I want the code output from template rendering to be autoformatted.

Even if the code was autoformatted *before* template rendering, the templating operations might have changed that. The line breaks or indentation might be gotten weird after string-replacement happened.

Maybe we could run autoformatters on the scratch directory before we output it. Perhaps the template spec.yaml could opt out of this behavior for the rare case where this isn't appropriate.

Open questions:
 - Can we just assume that formatting tools are present when running the CLI?
   - Are they in the $PATH?
   - Should we prompt the user to install missing autoformatting tools? (e.g. `go install mvdan.cc/gofumpt@latest`)

**Comment by capri-xiyue** [comment]
I wrote a design doc. But during the design discussion, we found that it looks like abc cli is used for sharing code across teams instead of within teams. (For example, the use case like this is rare. One org called "A" wants to write a template and the code output should match the code style "A" org wants and such template won't be used outside of "A" org). As abc cli is just used for sharing code across teams, it makes sense that template authors should not configure format style in abc cli as different template users probably have different code style preferences. At the same time, previously format tool was preferred in abc cli is because string replacement operations might make the code ugly and Dave just want abc cli to output nice code. Since the nice code abc cli thinks probably won't be the code style end users wants. It may make more sense that abc cli doesn't format the code and let future workflow engine to do such code format tasks.
